wifi: Upmerge 2 (fixes only) (nrf_wifi) (#21296)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Wi-Fi drivers
-------------

> * Added:

Reviewer: Please remove the changelog entry from this PR, and add it as a suggestion in https://github.com/nrfconnect/sdk-nrf/pull/21339.

Wi-Fi drivers
-------------

* Added:

>   * Advanced debug shell for reading and writing registers/memory of the nRF70 Series chip.

Reviewer: ```suggestion
  * Advanced debug shell for reading and writing registers and memory of the nRF70 Series chip.
```

    The debug shell can be enabled using the :kconfig:option:`CONFIG_WIFI_NRF70_DEBUG_SHELL` Kconfig option.

>   * A new shell command ``nrf70 util rpu_stats_mem`` is added to retrieve the RPU statistics even when the RPU processors are not functional (e.g., LMAC processor crashed).

Reviewer: ```suggestion
  * A new shell command, ``nrf70 util rpu_stats_mem`` is added to retrieve the RPU statistics even when the RPU processors are not functional (for example, if the LMAC processor has crashed).
```
